src/attributedlabel/unittests/NIAttributedLabelTests.m

```diff
@@ -16,18 +16,97 @@
 
 // See: http://bit.ly/hS5nNh for unit test macros.
 
+#import <Foundation/Foundation.h>
 #import <XCTest/XCTest.h>
 
 #import "NimbusAttributedLabel.h"
+#import "NIAttributedLabel+Testing.h"
+
+@interface NIAttributedLabelTestMonitor : NSObject <NIAttributedLabelDelegate>
+
+@property(nonatomic, assign) BOOL doNotRespondToLongPressCallback;
+
+@property(nonatomic, readonly) int longPressCallbackCount;
+@property(nonatomic, readonly) int actionSheetCallbackCount;
+
+@end
+
+@implementation NIAttributedLabelTestMonitor
+
+- (BOOL)respondsToSelector:(SEL)selector {
+  if (selector == @selector(attributedLabel:didLongPressTextCheckingResult:atPoint:)) {
+    return !_doNotRespondToLongPressCallback;
+  }
+
+  return [super respondsToSelector:selector];
+}
+
+- (void)attributedLabel:(NIAttributedLabel *)attributedLabel didLongPressTextCheckingResult:(NSTextCheckingResult *)result atPoint:(CGPoint)point {
+  _longPressCallbackCount++;
+}
+
+- (BOOL)attributedLabel:(NIAttributedLabel *)attributedLabel shouldPresentActionSheet:(UIActionSheet *)actionSheet withTextCheckingResult:(NSTextCheckingResult *)result atPoint:(CGPoint)point {
+  _actionSheetCallbackCount++;
+  return NO;
+}
+
+@end
 
 @interface NIAttributedLabelTests : XCTestCase
 @end
 
 
 @implementation NIAttributedLabelTests
 
+- (void)testLongPressCallbackOverridesActionSheetCallbackWhenTouchingLink {
+  // Create an attributed string with a link.
+  NSString *string = @"NimbusKit";
+  NSURL *URL = [NSURL URLWithString:@"http://nimbuskit.info/"];
+  NSTextCheckingResult *textCheckingResult = [NSTextCheckingResult linkCheckingResultWithRange:NSMakeRange(0, string.length) URL:URL];
+  NSDictionary<NSAttributedStringKey,id> *attributes = @{ NIAttributedLabelLinkAttributeName : textCheckingResult };
```

Comment:
nit: add a space after the comma in the lightweight generics

Reply:
Done